src/core/Config.cpp

```diff
@@ -63,6 +64,19 @@ void Config::sync()
     m_settings->sync();
 }
 
+void Config::upgrade()
+{
+    for (auto setting : deprecationMap.keys()) {
+        if (m_settings->contains(setting)) {
+            if(deprecationMap.value(setting) != "REMOVED") {
```

Comment:
I do not like "magic values". I would prefer it to just test for empty (ie. !deprecationMap.value(setting).isEmpty()) which means it was deleted.